gems/sorbet-runtime/lib/types/props/decorator.rb

```diff
@@ -156,10 +156,27 @@ def validate_prop_value(prop, val)
       else
         type_object.valid?(val)
       end
+
     if !valid
       raise T::Props::InvalidValueError.new("Can't set #{@class.name}.#{prop} to #{val.inspect} " \
         "(instance of #{val.class}) - need a #{type_object}")
     end
+  rescue T::Props::InvalidValueError => err
+    caller_loc = T.must(caller_locations(8, 1))[0]
+
+    pretty_message = "Parameter '#{prop}': #{err.message}\n" \
+      "Caller: #{caller_loc.path}:#{caller_loc.lineno}\n"
+
+    T::Configuration.call_validation_error_handler(
+      rules,
```

Comment:
Rather than the rules, would you be ok with passing `nil` here? `Signature` is a relatively well-defined / stable interface, but `rules` is completely internal and has never been exposed publicly before.

```suggestion
      nil,
```